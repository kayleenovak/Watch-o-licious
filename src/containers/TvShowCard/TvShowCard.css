.tv-show-card {
  width: 275px;
  height: 405px;
}

.tv-card-title {
  width: 210px;
  font-family: 'Montserrat', sans-serif;
  margin-bottom: 7px;
  font-size: 17px;
  color: white;
}

.tv-card-img {
  box-shadow: 2px 2px 10px black;
  border-radius: 20px;
  position: relative;
  width: 210px;
}

.tv-card-image {
  border-radius: 20px;
}

.tv-card-image:hover {
  opacity: .5;
}

.view-series-btn:hover .tv-card-image {
  opacity: .5;
}

.tv-card-network {
  font-family: 'Raleway', sans-serif;
  font-size: 14px;
  margin: 0;
  color: #aea7b9;
}

