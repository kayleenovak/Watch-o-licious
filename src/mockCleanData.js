export const mockCleanCakeShows = [
  {
    id: 921,
    title: 'Cake Boss',
    runtime: 30,
    premiered: '2009-05-25',
    schedule: {
      'time': '21:00',
      'days': [
        'Monday'
      ]
    },
    rating: null,
    network: 'TLC',
    summary: 'Cake Boss is a popular docu-series following beloved baker Buddy Valastro, his larger-than-life family and burgeoning businesses.',
    image: "http://static.tvmaze.com/uploads/images/medium_portrait/6/16129.jpg",
    largeImage: "http://static.tvmaze.com/uploads/images/original_untouched/6/16129.jpg"
  },
  {
    id: 27786,
    title: 'Extreme Cake Makers',
    runtime: 30,
    premiered: '2017-05-01',
    schedule: {
        time: "17:30",
        days: [
          "Monday",
          "Tuesday",
          "Wednesday",
          "Thursday",
          "Friday"
      ]
    },
    rating: null,
    network: 'Channel 4',
    summary: "Welcome to the world of extreme cake making, where sugar craft specialists up and down the country dream up ever more elaborate ways to satisfy our sweet tooth.",
    image: "http://static.tvmaze.com/uploads/images/medium_portrait/110/275113.jpg",
    largeImage: "http://static.tvmaze.com/uploads/images/original_untouched/110/275113.jpg"
  }
]

export const mockCleanBakeShows = [
    {
    id: 2950,
    title: 'The Great British Bake Off',
    runtime: 75,
    premiered: '2010-08-17',
    schedule: {
      time: "20:00",
      days: [
        "Tuesday"
      ]
    },
    rating: 8.7,
    network: 'Channel 4',
    summary: "Over 10 weeks, 12 of the best amateur bakers in Britain will whisk, knead, ice, beat and bake their way through classic British cakes, perfect patisserie, Italian delights, sticky caramel constructions and elaborate layered puddings. All 12 will be hoping to impress with their skill, creativity, knowledge and passion to clinch the Bake Off Crown. Each of the 30 new challenges have been carefully designed by judges Paul Hollywood and Prue Leith to reveal just who is a star baker. But it's not just a new experience for the bakers. Also joining the tent for the first time are Sandi Toksvig and Noel Fielding, who will be with the bakers every step of the way, ready with a pertinent pep talk, a helpful hand or just a sympathetic shoulder to cry on.",
    image: "http://static.tvmaze.com/uploads/images/medium_portrait/124/311372.jpg",
    largeImage: "http://static.tvmaze.com/uploads/images/original_untouched/124/311372.jpg"
  },
    {
    id: 39695,
    title: 'The Great Kiwi Bake Off',
    runtime: 60,
    premiered: '2018-10-16',
    schedule: {
      time: "19:30",
      days: [
        "Tuesday"
      ]
    },
    rating: null,
    network: 'TVNZ 2',
    summary: "From biscuits to bread, pies to pavlovas, 12 amateur Kiwi bakers have their skills put to the test. Will they rise to the challenge?",
    image: "http://static.tvmaze.com/uploads/images/medium_portrait/174/435768.jpg",
    largeImage: "http://static.tvmaze.com/uploads/images/original_untouched/174/435768.jpg"
  }
]

export const mockCleanBakingShows = [
  {
    id: 1549,
    title: 'Kids Baking Championship',
    runtime: 60,
    premiered: '2015-02-02',
    schedule: {
      time: "21:00",
      days: [
          "Monday"
      ]
    },
    rating: null,
    network: 'Food Network',
    summary: "Hosts and judges Duff Goldman and Valerie Bertinelli lead eight talented kid bakers through tasty challenges designed to find the most impressive and creative young baker on Kids Baking Championship. In each challenge, the kid contestants (ranging in age from 10 to 13) will be tested on their baking skills and originality, as they whip up delectable desserts including creative cakes, mouthwatering cookies and pastries making it seem as easy as pie. At stake, a sweet prize package that includes a complete set of Food Network bakeware products, a profile in Food Network Magazine, an original cake featured at Charm City Cakes and a $10,000 grand prize. Only one will take the cake to become the Kids Baking Champion!",
    image: "http://static.tvmaze.com/uploads/images/medium_portrait/88/220914.jpg",
    largeImage: "http://static.tvmaze.com/uploads/images/original_untouched/88/220914.jpg"
  },
  {
    id: 6407,
    title: 'Holiday Baking Championship',
    runtime: 60,
    premiered: '2014-11-09',
    schedule: {
      time: "21:00",
      days: [
        "Monday"
      ]
    },
    rating: null,
    network: 'Food Network',
    summary: "On Holiday Baking Championship the search to find the greatest holiday baker begins as eight amateur bakers enter the kitchen, where they show off their family traditions and superb baking skills. To survive the challenges from week to week, they must prove their abilities in front of our tough-love judges: Duff Goldman, Nancy Fuller and Lorraine Pascale. Hosted by Bobby Deen, this baking championship is the sweetest way to celebrate this holiday season.",
    image: "http://static.tvmaze.com/uploads/images/medium_portrait/24/61290.jpg",
    largeImage: "http://static.tvmaze.com/uploads/images/original_untouched/24/61290.jpg"
  }
]