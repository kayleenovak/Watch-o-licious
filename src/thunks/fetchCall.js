export const fetchCall = async (url) => {
  console.log(url)
  const response = await fetch(url)
  if (!response.ok) {
    throw Error(response.statusText)
  }
  const tvShows = await response.json()
  return tvShows
}