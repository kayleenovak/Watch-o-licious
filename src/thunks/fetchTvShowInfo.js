import { isLoading, hasErrored, tvShowFetchSuccess } from '../actions/index.js'
import { tvShowCleaner } from '../cleaners/tvShowCleaner.js'

export const fetchTvShowInfo = (id) => {
  const tvShowUrl = `http://api.tvmaze.com/shows/${id}/episodes`
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const tvShow = await fetchCall(tvShowUrl)
      dispatch(isLoading(false))
      const cleanedShow = await tvShowCleaner(tvShow)
      dispatch(tvShowFetchSuccess(cleanedShow))
    } catch (error) {
      dispatch(hasErrored(true))
    }
  }
}