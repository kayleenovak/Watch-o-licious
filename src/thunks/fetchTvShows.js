import { isLoading, hasErrored, dataFetchSuccess } from '../actions/index.js'
import { tvShowsCleaner } from '../cleaners/tvShowsCleaner.js'

export const fetchTvShows = (url) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url)
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const tvShows = await response.json()
      const cleanedTvShows = await tvShowsCleaner(tvShows)
      dispatch(dataFetchSuccess(cleanedTvShows))
    } catch (error) {
      dispatch(hasErrored(true))
    }
  }
}