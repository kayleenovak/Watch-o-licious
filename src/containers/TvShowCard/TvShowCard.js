import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { fetchTvShowInfo } from '../../thunks/fetchTvShowInfo.js'
import './TvShowCard.css'
import { Link } from 'react-router-dom'

export const TvShowCard = (props) => {
  const { title, network, image, id } = props
  return (
    <article className='tv-show-card'>
      <Link to={`/tvshow/${id}`} key={id}>
        <div className='tv-card-img'>
          <img alt='series poster' src={image} className='tv-card-image'/>
        </div>
      </Link>
      <h3 className='tv-card-title'>{title}</h3>
      <p className='tv-card-network'>Network: {network}</p>
    </article>
  )
}

export const mapDispatchToProps = (dispatch) => ({
  fetchShowInfo: (id) => dispatch(fetchTvShowInfo(id))
})

export default connect(null, mapDispatchToProps)(TvShowCard)

TvShowCard.propTypes = {
  fetchShowInfo: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  network: PropTypes.string.isRequired,
  image: PropTypes.string,
  id: PropTypes.number.isRequired,
}
