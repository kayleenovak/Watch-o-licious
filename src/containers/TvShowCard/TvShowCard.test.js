import React from 'react'
import { TvShowCard, mapDispatchToProps } from './TvShowCard.js'
import { mockCleanCakeShows } from '../../mockCleanData.js'
import { fetchTvShowInfo } from '../../thunks/fetchTvShowInfo'
import { shallow } from 'enzyme'

jest.mock('../../thunks/fetchTvShowInfo')

describe('TvShowCard', () => {
  it('should return an article with an image, h3, 2 p elements and to match the snapshot', () => {
    const mockFetch = jest.fn()
    const wrapper = shallow(<TvShowCard title={'Cake Boss'} network={'FOX'} img={'www.img.com'} id={1} fetchShowInfo={ mockFetch }/>)

    expect(wrapper.find('article').length).toEqual(1)
    expect(wrapper.find('h3').length).toEqual(1)
    expect(wrapper.find('img').length).toEqual(1)
    expect(wrapper.find('p').length).toEqual(1)
    expect(wrapper).toMatchSnapshot()
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the correct params', () => {
      const mockDispatch = jest.fn()
      const mockId = 1

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.fetchShowInfo(mockId)

      expect(mockDispatch).toHaveBeenCalledWith(fetchTvShowInfo(mockId))
    })
  })  
})