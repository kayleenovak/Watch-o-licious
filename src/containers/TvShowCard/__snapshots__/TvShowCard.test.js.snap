// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TvShowCard should return an article with an image, h3, 2 p elements and to match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TvShowCard
    fetchShowInfo={[MockFunction]}
    id={1}
    img="www.img.com"
    network="FOX"
    title="Cake Boss"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/tvshow/1"
        >
          <div
            className="tv-card-img"
          >
            <img
              alt="series poster"
              className="tv-card-image"
            />
          </div>
        </Link>,
        <h3
          className="tv-card-title"
        >
          Cake Boss
        </h3>,
        <p
          className="tv-card-network"
        >
          Network: 
          FOX
        </p>,
      ],
      "className": "tv-show-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="tv-card-img"
          >
            <img
              alt="series poster"
              className="tv-card-image"
            />
          </div>,
          "replace": false,
          "to": "/tvshow/1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="series poster"
              className="tv-card-image"
            />,
            "className": "tv-card-img",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "series poster",
              "className": "tv-card-image",
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Cake Boss",
          "className": "tv-card-title",
        },
        "ref": null,
        "rendered": "Cake Boss",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Network: ",
            "FOX",
          ],
          "className": "tv-card-network",
        },
        "ref": null,
        "rendered": Array [
          "Network: ",
          "FOX",
        ],
        "type": "p",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/tvshow/1"
          >
            <div
              className="tv-card-img"
            >
              <img
                alt="series poster"
                className="tv-card-image"
              />
            </div>
          </Link>,
          <h3
            className="tv-card-title"
          >
            Cake Boss
          </h3>,
          <p
            className="tv-card-network"
          >
            Network: 
            FOX
          </p>,
        ],
        "className": "tv-show-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "children": <div
              className="tv-card-img"
            >
              <img
                alt="series poster"
                className="tv-card-image"
              />
            </div>,
            "replace": false,
            "to": "/tvshow/1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="series poster"
                className="tv-card-image"
              />,
              "className": "tv-card-img",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "series poster",
                "className": "tv-card-image",
                "src": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Cake Boss",
            "className": "tv-card-title",
          },
          "ref": null,
          "rendered": "Cake Boss",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Network: ",
              "FOX",
            ],
            "className": "tv-card-network",
          },
          "ref": null,
          "rendered": Array [
            "Network: ",
            "FOX",
          ],
          "type": "p",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
